modules/lang-painless/src/main/java/org/elasticsearch/painless/MethodWriter.java

```diff
@@ -200,9 +200,15 @@ private void writeCast(final Type from, final Type to) {
         }
     }
 
+    /**
+     * ASM does boxing in an evil way to be compatible with Java prior to verison 5.
+     * Supporting versions prior to 5 is not a requirement of this project, so the modern
+     * boxing methods are used instead.
+     */
     @Override
     public void box(final org.objectweb.asm.Type type) {
         switch (type.getSort()) {
+            case org.objectweb.asm.Type.VOID:    visitInsn(Opcodes.ACONST_NULL);                     break;
```

Comment:
I'd remove this again. Because this is done in original ASM, just to prevent incorrect stack on voids